scipy/optimize/_linprog_highs.py

```diff
@@ -13,108 +13,197 @@
+            ),
+        }
+
+    def get_scipy_status(self, highs_status, highs_message):
+        """Converts HiGHS status and message to SciPy-like status and messages"""
+        if highs_status is None or highs_message is None:
+            print(f"Highs Status: {highs_status}, Message: {highs_message}")
+            return (
+                self.SciPyRC.NUMERICAL.value,
+                "HiGHS did not provide a status code. (HiGHS Status None: None)",
+            )
+
+        scipy_status_enum, message_prefix, scipy_message = self.highs_to_scipy.get(
+            _h.HighsModelStatus(highs_status),
+            (
+                self.SciPyRC.NUMERICAL,
+                "Unknown HiGHS Status",
+                "The HiGHS status code was not recognized.",
+            ),
+        )
+        full_scipy_message = (
+            f"{scipy_message} (HiGHS Status {int(highs_status)}: {highs_message})"
+        )
+        return scipy_status_enum.value, full_scipy_message
 
 
 def _replace_inf(x):
-    # Replace `np.inf` with CONST_INF
+    # Replace `np.inf` with kHighsInf
     infs = np.isinf(x)
     with np.errstate(invalid="ignore"):
-        x[infs] = np.sign(x[infs])*CONST_INF
+        x[infs] = np.sign(x[infs]) * _h.kHighsInf
     return x
 
 
-def _convert_to_highs_enum(option, option_str, choices):
-    # If option is in the choices we can look it up, if not use
-    # the default value taken from function signature and warn:
+class SimplexStrategy(Enum):
+    DANTZIG = "dantzig"
+    DEVEX = "devex"
+    STEEPEST_DEVEX = "steepest-devex"  # highs min, choose
+    STEEPEST = "steepest"  # highs max
+
+    def to_highs_enum(self):
+        mapping = {
+            SimplexStrategy.DANTZIG: simpc.SimplexEdgeWeightStrategy.kSimplexEdgeWeightStrategyDantzig.value,
+            SimplexStrategy.DEVEX: simpc.SimplexEdgeWeightStrategy.kSimplexEdgeWeightStrategyDevex.value,
+            SimplexStrategy.STEEPEST_DEVEX: simpc.SimplexEdgeWeightStrategy.kSimplexEdgeWeightStrategyChoose.value,
+            SimplexStrategy.STEEPEST: simpc.SimplexEdgeWeightStrategy.kSimplexEdgeWeightStrategySteepestEdge.value,
+        }
+        return mapping.get(self)
+
+
+def convert_to_highs_enum(option, option_str, choices_enum, default_value):
+    if option is None:
+        return choices_enum[default_value.upper()].to_highs_enum()
     try:
-        return choices[option.lower()]
-    except AttributeError:
-        return choices[option]
+        enum_value = choices_enum[option.upper()]
     except KeyError:
-        sig = inspect.signature(_linprog_highs)
-        default_str = sig.parameters[option_str].default
-        warn(f"Option {option_str} is {option}, but only values in "
-             f"{set(choices.keys())} are allowed. Using default: "
-             f"{default_str}.",
-             OptimizeWarning, stacklevel=3)
-        return choices[default_str]
-
-
-def _linprog_highs(lp, solver, time_limit=None, presolve=True,
```

Comment:
A large majority of lines in the diff from here on down in this file are style chances (single to double quotes, etc.). It's now hard to disentangle that, but for future reference: please avoid mixing in style changes, especially very minor ones that don't really improve readability - it makes reviewing harder.

Reply:
Yup, sorry about that, I reversed the formatting changes so it should be clearer now.